mmdet/core/bbox/assigners/center_region_assigner.py

```diff
@@ -0,0 +1,297 @@
+import torch
+
+from ..builder import BBOX_ASSIGNERS
+from ..iou_calculators import build_iou_calculator
+from .assign_result import AssignResult
+from .base_assigner import BaseAssigner
+
+
+def scale_boxes(bboxes, scale):
+    """Expand an array of boxes by a given scale.
+        Args:
+            bboxes (Tensor): shape (m, 4)
+            scale (float): the scale factor of bboxes
+
+        Returns:
+            (Tensor): shape (m, 4) scaled bboxes
+        """
+    w_half = (bboxes[:, 2] - bboxes[:, 0]) * .5
+    h_half = (bboxes[:, 3] - bboxes[:, 1]) * .5
+    x_c = (bboxes[:, 2] + bboxes[:, 0]) * .5
+    y_c = (bboxes[:, 3] + bboxes[:, 1]) * .5
+
+    w_half *= scale
+    h_half *= scale
+
+    boxes_scaled = torch.zeros_like(bboxes)
+    boxes_scaled[:, 0] = x_c - w_half
+    boxes_scaled[:, 2] = x_c + w_half
+    boxes_scaled[:, 1] = y_c - h_half
+    boxes_scaled[:, 3] = y_c + h_half
+    return boxes_scaled
+
+
+def is_located_in(points, bboxes, is_aligned=False):
+    """ is center a locates in box b
+    Then we compute the area of intersect between box_a and box_b.
+    Args:
+      points: (tensor) bounding boxes, Shape: [m,2].
+      bboxes: (tensor)  bounding boxes, Shape: [n,4].
+       If is_aligned is ``True``, then m mush be equal to n
+    Return:
+      (tensor) intersection area, Shape: [m, n]. If is_aligned ``True``,
```

Comment:
`Tensor: intersection area, shape (m, n) if ``is_aligned `` is ``False``, else shape (m, )`